src/types/typeguards.ts

```diff
@@ -1,14 +1,29 @@
+import { Update } from "@grammyjs/types";
 import { Context } from "./context";
+import { PluginInputs } from "./plugin-inputs";
 
-/**
- * Typeguards are most helpful when you have a union type and you want to narrow it down to a specific one.
- * In other words, if `SupportedEvents` has multiple types then these restrict the scope
- * of `context` to a specific event payload.
- */
-
-/**
- * Restricts the scope of `context` to the `issue_comment.created` payload.
- */
-export function isIssueCommentEvent(context: Context): context is Context<"issue_comment.created"> {
-  return context.eventName === "issue_comment.created";
+export function isIssueOpenedEvent(context: Context): context is Context<"issues.opened"> {
+  return context.eventName === "issues.opened";
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isTelegramPayload(payload: any): payload is Update {
```

Comment:
Maybe the types should be renamed to say telegram

Reply:
use `unknown` type if is it's really unknown because it makes you check the object shape but `any` will allow you to do anything so it's more typesafe than `any`

Reply:
I went with the choice below. `GrammyTelegramUpdate` is just an alias type for the package export type `Update`

```ts
export function isTelegramPayload(payload: unknown): payload is GrammyTelegramUpdate {
  if (typeof payload !== "object" || !payload) return false;
  return "update_id" in payload && payload.update_id !== undefined;
}
```